rust/main/hyperlane-base/src/settings/chains.rs

```diff
@@ -183,6 +183,7 @@ impl ChainConf {
             ChainConnectionConf::Sealevel(conf) => Ok(Box::new(h_sealevel::SealevelProvider::new(
                 locator.domain.clone(),
                 conf,
+                self.addresses.mailbox,
```

Comment:
worth documenting why we do this imo, otherwise this could be easily forgotten

Reply:
I added comments.